benchmarks/osb/README.md

```diff
@@ -0,0 +1,399 @@
+
+OpenSearch Benchmark is a framework for performance benchmarking an OpenSearch 
+cluster. For more details, checkout their 
+[repo](https://github.com/opensearch-project/opensearch-benchmark/). 
+
+Before getting into the benchmarks, it is helpful to know a few terms:
+1. Workload - Top level description of a benchmark suite. A workload will have a `workload.json` file that defines different components of the tests 
+2. Test Procedures - A workload can have a schedule of operations that run the test. However, a workload can also have several test procedures that define their own schedule of operations. This is helpful for sharing code between tests
+3. Operation - An action against the OpenSearch cluster
+4. Parameter source - Producers of parameters for OpenSearch operations
+5. Runners - Code that actually will execute the OpenSearch operations
+
+### Setup
+
+OpenSearch Benchmarks requires Python 3.8 or greater to be installed. One of 
+the easier ways to do this is through Conda, a package and environment 
+management system for Python.
+
+First, follow the 
+[installation instructions](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) 
+to install Conda on your system.
+
+Next, create a Python 3.8 environment:
+```
+conda create -n knn-osb python=3.8
+```
+
+After the environment is created, activate it:
+```
+source activate knn-osb
+```
+
+Lastly, clone the k-NN repo and install all required python packages:
+```
+git clone https://github.com/opensearch-project/k-NN.git
+cd k-NN/benchmarks/osb
+pip install -r requirements.txt
+```
+
+After all of this completes, you should be ready to run your first benchmark!
+
+### Running a benchmark
+
+Before running a benchmark, make sure you have the endpoint of your cluster and
+ and the machine you are running the benchmarks from can access it. 
+ Additionally, ensure that all data has been pulled to the client.
+
+Currently, we support 2 test procedures for the k-NN workload: train-test and 
+no-train-test. The train test has steps to train a model included in the 
+schedule, while no train does not. Both test procedures will index a data set 
+of vectors into an OpenSearch index. 
+
+Once you have decided which test procedure you want to use, open up 
+[params/train-params.json](params/train-params.json) or 
+[params/no-train-params.json](params/no-train-params.json) and 
+fill out the parameters. Notice, at the bottom of `no-train-params.json` there 
+are several parameters that relate to training. Ignore these. They need to be 
+defined for the workload but not used. 
+
+Once the parameters are set, set the URL and PORT of your cluster and run the 
+command to run the test procedure. 
+
+```
+export URL=
+export PORT=
+export PARAMS_FILE=
+export PROCEDURE={no-train-test | train-test}
+
+opensearch-benchmark execute_test \ 
+    --target-hosts $URL:$PORT \ 
+    --workload-path ./workload.json \ 
+    --track-params ${PARAMS_FILE} \
```

Comment:
I think this should be `--workload-params` instead

Reply:
Good catch, fixed